packages/dappmanager/src/utils/portsTableData.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  InstalledPackageData,
+  PackagePort,
+  PortsTable,
+  TcpPortScan,
+  UpnpPortMapping
+} from "../common";
+import * as db from "../db";
+
+/**
+ * Returns the necessary data for display the
+ * ports table in the UI well formatted
+ */
+export function portsTableData({
+  apiTcpPortsStatus,
+  packages,
+  upnpPortMappings,
+  portsToOpen
+}: {
+  apiTcpPortsStatus: TcpPortScan[];
+  packages: InstalledPackageData[];
+  upnpPortMappings: UpnpPortMapping[];
+  portsToOpen: PackagePort[];
+}): PortsTable[] {
+  const upnpAvailable = db.upnpAvailable.get();
+
+  // COOKING DATA
+  return portsToOpen.map(port => {
+    return {
+      port: port.portNumber,
+      protocol: port.protocol,
+      upnpStatus: getUpnpStatus({ port, upnpAvailable, upnpPortMappings }),
+      apiStatus: getApiStatus({ port, apiTcpPortsStatus }),
+      service: getServiceName({ port, packages })
```

Comment:
You should use strictly one this "name" entities: `dnpName`, `serviceName`, `containerName`. Otherwise things get confusing. Note that `serviceName` does not uniquely identify a service globally. You should return `dnpName`, `serviceName` from the dappmanager and the UI should print that in a pretty way